### PR TITLE
Cleanups.

### PR DESCRIPTION
**Pull request by christianhaeubl**
I removed all TODOs from the sources. We should probably collect the TODOs listed below in a separate issue:
- Rewrite some of the Java-level JFR events as Native events (e.g., JavaThreadStatistics)
- JfrChunkWriter.openFile should probably write all live threads
- JfrNativeEventWriter.accommodate can lose the current event if the new buffer is null
- JfrThreadLocal.beforeThreadRun - parentThread is null at the moment
- JfrThreadLocal.beforeThreadRun - write a thread checkpoint so that JFR knows about the newly started thread
- JfrTypeRepository - fix the 'exported' field
- TestSingleEvent - parse & verify the constant pool

### CODE REVIEW

**Comment by christianhaeubl** [comment]
Which command line do you currently use for executing the unit tests?

**Comment by jiekang** [comment]
> Which command line do you currently use for executing the unit tests?

I think it's: `mx native-unittest com.oracle.svm.jfrtest --build-args -H:+AllowVMInspection`

File: substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrGlobalMemory.java [hunk 1]
```diff
@@ -141,13 +142,13 @@ private static void releasePromotionBuffer(JfrBuffer buffer) {
...
+        // Currently unused as in memory recording is not supported yet.
+        return false;
     }
 
     @Uninterruptible(reason = "Epoch must not change while in this method.")
-    private static boolean shouldDiscard() {
-        // TODO: implement
-        return false;
+    private static void discardOldest() {
+        VMError.unimplemented();
```
Comment (jiekang) [inline substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrGlobalMemory.java]: Looks like `VMError.unimplemented()` is not uninterruptible, so maybe temporarily set callee to be false?

**Review by jiekang — CHANGES_REQUESTED** [review]
One comment for `JfrGlobalMemory.shouldDiscard()`

There is also a failure on running atm with
```
No Java synchronization must be performed within a VMOperation: if the object is already locked, the VM is deadlocked
Fatal error: Should not reach here: Not okay to block.
```

Maybe after reworking on top of the chunk rotation that has the uninterruptible writes it'll be okay?

**Comment by christianhaeubl** [comment]
Rebased and fixed the remaining issues. The unit tests pass now.
@rkennke : it seems weird that the method `JfrTraceIdLoadBarrier.clear()` is currently unused. Please have a brief look on that.

**Review by jiekang — APPROVED** [review]

**Comment by rkennke** [comment]
> Rebased and fixed the remaining issues. The unit tests pass now.
> @rkennke : it seems weird that the method `JfrTraceIdLoadBarrier.clear()` is currently unused. Please have a brief look on that.

Yeah I guess it should be called right after epoch switch.

**Pull request merged**